test/unit/iotagent_plugin-test.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2014 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-orion-pep
+ *
+ * fiware-orion-pep is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * fiware-orion-pep is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with fiware-orion-pep.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[daniel.moranjimenez@telefonica.com]
+ */
+
+'use strict';
+
+var serverMocks = require('../tools/serverMocks'),
+    proxyLib = require('../../lib/fiware-orion-pep'),
+    restPlugin = require('../../lib/plugins/restPlugin'),
+    keystonePlugin = require('../../lib/services/keystoneAuth'),
+    config = require('../../config'),
+    utils = require('../tools/utils'),
+    should = require('should'),
+    async = require('async'),
+    request = require('request');
+
+describe('IOT Agent Plugin tests', function() {
+    var proxy,
+        mockServer,
+        mockApp,
+        mockAccess,
+        mockAccessApp,
+        mockOAuth,
+        mockOAuthApp,
+        apiCases = [
+            ['POST', '/iot/devices', 'create'],
```

Comment:
After F2F discussion, the behaviour is not correct... Possible solutions:

* Having separated actions, "createServices", "createDevices" (thus the generic REST pluging could be use)
* Include the resource from URL in the information passed to the AC, thus the AC takes decissions based on action + resource